src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -164,606 +136,64 @@ void AsyncRPCOperation_sendmany::main() {
-    }
-    if (!spendable.LimitToAmount(targetAmount, dustThreshold, recipientPools_)) {
-        CAmount changeAmount{spendable.Total() - targetAmount};
-        std::string insufficientFundsMessage =
-            strprintf("Insufficient funds: have %s", FormatMoney(spendable.Total()));
-        if (changeAmount > 0 && changeAmount < dustThreshold) {
-            // TODO: we should provide the option for the caller to explicitly
-            // forego change (definitionally an amount below the dust amount)
-            // and send the extra to the recipient or the miner fee to avoid
-            // creating dust change, rather than prohibit them from sending
-            // entirely in this circumstance.
-            // (Daira disagrees, as this could leak information to the recipient
-            // or to an external viewing key holder.)
-            insufficientFundsMessage +=
-                strprintf(
-                    ", need %s more to avoid creating invalid change output %s (dust threshold is %s)",
-                    FormatMoney(dustThreshold - changeAmount),
-                    FormatMoney(changeAmount),
-                    FormatMoney(dustThreshold));
-        } else {
-            insufficientFundsMessage += strprintf(", need %s", FormatMoney(targetAmount));
-        }
-        bool isFromUa = std::holds_alternative<libzcash::UnifiedAddress>(ztxoSelector_.GetPattern());
-        throw JSONRPCError(
-                RPC_WALLET_INSUFFICIENT_FUNDS,
-                insufficientFundsMessage
-                + (allowTransparentCoinbase && ztxoSelector_.SelectsTransparentCoinbase() ? "." :
-                   "; note that coinbase outputs will not be selected if you specify "
-                   "ANY_TADDR or if any transparent recipients are included.")
-                + ((!isFromUa || strategy_.AllowLinkingAccountAddresses()) ? "" :
-                   " (This transaction may require selecting transparent coins that were sent "
-                   "to multiple Unified Addresses, which is not enabled by default because "
-                   "it would create a public link between the transparent receivers of these "
-                   "addresses. THIS MAY AFFECT YOUR PRIVACY. Resubmit with the `privacyPolicy` "
-                   "parameter set to `AllowLinkingAccountAddresses` or weaker if you wish to "
-                   "allow this transaction to proceed anyway.)"));
-    }
-
-    if (!(spendable.utxos.empty() || strategy_.AllowRevealedSenders())) {
-        throw JSONRPCError(
-            RPC_INVALID_PARAMETER,
-            "This transaction requires selecting transparent coins, which is "
-            "not enabled by default because it will publicly reveal transaction "
-            "senders and amounts. THIS MAY AFFECT YOUR PRIVACY. Resubmit "
-            "with the `privacyPolicy` parameter set to `AllowRevealedSenders` "
-            "or weaker if you wish to allow this transaction to proceed anyway.");
-    }
```

Comment:
This is now covered by the more generic `GetRequiredPrivacyPolicy` check. Later PRs in the WalletTxBuilder DAG make that check richer, explaining the factors contributing to the required policy, rather than something along the lines of “the provided policy is not compatible with the required policy.”